subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/maven/MavenDynamicResolveIntegrationTest.groovy

```diff
@@ -376,24 +376,25 @@ Searched in the following locations:
         when:
         repo1.getModuleMetaData("group", "projectA").expectGet()
         projectA1.pom.expectGet()
-        projectA1.getArtifact().expectGet()
 
         repo2.getModuleMetaData("group", "projectA").expectGet()
         projectA2.pom.expectGetBroken()
```

Comment:
❌ Can you make sure we still have test coverage for the "missing" case? Otherwise we might be breaking behavior here.

Reply:
@blindpirate I think we should add a test like this:

```
    def "dynamic version ignores missing module in one repository when available in another repository"()  {
        given:
        def repo1 = mavenHttpRepo("repo1")
        def repo2 = mavenHttpRepo("repo2")

        def projectA2 = repo2.module('group', 'projectA', '1.5').publish()

        buildFile << """
        repositories {
            maven { url '${repo1.uri}' }
            maven { url '${repo2.uri}' }
        }
        configurations { compile }
        dependencies {
            compile 'group:projectA:1.+'
        }

        task retrieve(type: Sync) {
            into 'libs'
            from configurations.compile
        }
        """

        when:
        repo1.getModuleMetaData("group", "projectA").expectGetMissing()
        repo1.directory("group", "projectA").expectGetMissing()
        
        repo2.getModuleMetaData("group", "projectA").expectGet()
        projectA2.pom.expectGet()
        projectA2.artifact.expectGet()

        then:
        succeeds 'retrieve'
    }
```